src/workerd/jsg/jsg.h

```diff
@@ -2829,6 +2846,12 @@ class Lock {
   // This variation includes modules from the worker bundle.
   kj::Maybe<JsObject> resolveModule(kj::StringPtr specifier);
 
+  // Used for the underlying implementation of the file system API
+  // Only file:// URLs are supported. If the target does not exist
+  // and createAs is not set, kj::none is returned. If createAs is
+  // set, the target is created as the given type then returned.
+  kj::Maybe<kj::OneOf<kj::Own<fs::File>, kj::Own<fs::Directory>>> resolveVfsNode(const Url& path);
```

Comment:
Since both fs::File and fs::Directory inherit from fs::Node, why not just return fs::Node?

Reply:
I'll just remove `fs::Node`. Stylistically I have a preference towards using `KJ_SWITCH_ONEOF` rather than casting from `Node`.